samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -35,23 +37,43 @@
  *
  * Callbacks issued from  {@link ClusterResourceManager} aka {@link ContainerProcessManager} are intercepted
  * by ContainerManager to handle container failure and completions for both active and standby containers
+ *
+ * ContainerManager encapsulates logic and state related to container placement actions like move, restarts for active container
+ * if issued externally.
+ *
+ * TODO SAMZA-2378: Container Placements for Standby containers enabled jobs
+ *      SAMZA-2379: Container Placements for job running in degraded state
  */
 public class ContainerManager {
 
   private static final Logger LOG = LoggerFactory.getLogger(ContainerManager.class);
+  private static final String ANY_HOST = ResourceRequestState.ANY_HOST;
 
+  /**
+   * Default timeout for request to the cluster manager for stopping an active container & checking expiry for
+   * requested resources
+   */
+  private static final Long DEFAULT_CONTROL_ACTION_EXPIRY = Duration.ofSeconds(10).toMillis();
   /**
    * Resource-manager, used to stop containers
    */
   private final ClusterResourceManager clusterResourceManager;
   private final SamzaApplicationState samzaApplicationState;
+  private final boolean hostAffinityEnabled;
+
+  /**
+   * Map maintaining active container placement action meta data indexed by containers processorId eg 0, 1, 2
+   */
+  private final ConcurrentHashMap<String, ContainerPlacementMetadata> actions;
```

Comment:
Please add comments here to indicate, why it is indexed by processorId and not containerID?
Is it to ensure there is a single request inflight per processorId at all times?